pkg/operator/certrotation/client_cert_rotation_controller.go

```diff
@@ -117,97 +116,24 @@ func (c CertRotationController) syncWorker() error {
 	return nil
 }
 
-func (c *CertRotationController) WaitForReady(stopCh <-chan struct{}) {
-	klog.Infof("Waiting for CertRotationController - %q", c.name)
-	defer klog.Infof("Finished waiting for CertRotationController - %q", c.name)
-
-	if !cache.WaitForCacheSync(stopCh, c.cachesToSync...) {
-		utilruntime.HandleError(fmt.Errorf("caches did not sync"))
-		return
+func (c CertRotationController) targetCertRecheckerPostRunHook(ctx context.Context, syncCtx factory.SyncContext) error {
+	// If we have a need to force rechecking the cert, use this channel to do it.
+	refresher, ok := c.TargetRotation.CertCreator.(TargetCertRechecker)
+	if !ok {
+		return nil
 	}
-}
-
-// RunOnce will run the cert rotation logic, but will not try to update the static pod status.
-// This eliminates the need to pass an OperatorClient and avoids dubious writes and status.
-func (c *CertRotationController) RunOnce() error {
-	return c.syncWorker()
-}
-
-func (c *CertRotationController) Run(ctx context.Context, workers int) {
-	defer utilruntime.HandleCrash()
-	defer c.queue.ShutDown()
-
-	klog.Infof("Starting CertRotationController - %q", c.name)
-	defer klog.Infof("Shutting down CertRotationController - %q", c.name)
-	c.WaitForReady(ctx.Done())
-
-	// doesn't matter what workers say, only start one.
-	go wait.UntilWithContext(ctx, c.runWorker, time.Second)
-
-	// start a time based thread to ensure we stay up to date
+	targetRefresh := refresher.RecheckChannel()
 	go wait.Until(func() {
-		ticker := time.NewTicker(time.Minute)
-		defer ticker.Stop()
-
 		for {
-			c.queue.Add(workQueueKey)
 			select {
-			case <-ticker.C:
+			case <-targetRefresh:
+				syncCtx.Queue().Add(factory.QueueKey(c.name))
```

Comment:
why do we queue names?

Reply:
prometheus metrics? (each queue have one, makes it easier to see which queue is busy)

Reply:
but this is not the queue, it's the key in a queue. I hope this is not exposed as metrics label as that would be CVE worths (memory explosion attack)